### PR TITLE
Feature bomb2

### LINKED ISSUE #56: Implement bomb objects in the game
subtask of: #39

### PR DESCRIPTION
**Pull request by WytzeElhorst**
New branch to fix merge request

This closes #56 and #57

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.jme3.material.Material;
+import com.jme3.math.ColorRGBA;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Node;
+import com.jme3.scene.Spatial;
+import com.jme3.scene.shape.Sphere;
+
+/**
+ * Class representing the player wearing the VR headset.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java]: No it isn't.

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class Bomb extends Entity {
+	private Geometry geometry;
+	private Spatial sp;
+	/**
+	 * Constructor for a default player.
+	 * This player is (for now) a red sphere.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java]: Again, no it isn't.

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	@Override
+	public void simpleUpdate(float tdf) {
+		sp.move(0, 0, 0);
+	}
+
+	@Override
+	public void mapDraw(Graphics2D g, int resolution) {
+		Vector3f trans = geometry.getLocalTranslation();
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: This code is a duplicate from the VRPlayer class. Code should be reused.

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		Material matb = new Material(Main.getInstance().getAssetManager(), "Common/MatDefs/Misc/Unshaded.j3md");
+		matb.setColor("Color", ColorRGBA.White);
+		((Node) nod.getChild("Cylinder.001")).getChild(0).setMaterial(matb);
+		sp = nod;
+	}
+
+	@Override
+	public Geometry getGeometry() {
+		return geometry;
+	}
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:42]: There should be line breaks between different methods.

File: src/main/java/nl/tudelft/contextproject/Bomb.java (hunk 3 above)
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: @BCrlrd What do you mean? This looks fine to me.

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.awt.Graphics2D;
+
+import com.jme3.material.Material;
+import com.jme3.math.ColorRGBA;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Node;
+import com.jme3.scene.Spatial;
+import com.jme3.scene.shape.Sphere;
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java]: Unused import should be removed.

**Comment by Geweldig** [comment]
The class has 0% line coverage according to cobertura. This should be fixed.

File: src/main/java/nl/tudelft/contextproject/Bomb.java (hunk 3 above)
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: It's the exact same code as in the VRPlayer class. If these things act in the exact same way it's bad practice to duplicate it over two classes.

Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: @BCrlrd The code just happens to be the same at this point in time. VRPlayer and Bomb do different things. It is likely that VRPlayer's code for drawing and updating changes when controls are implemented, so I don't think that they should share the same base class with code.

Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: Alright, I can agree with this.

File: src/main/java/nl/tudelft/contextproject/Bomb.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	public Geometry getGeometry() {
+		return geometry;
+	}
+	public Spatial getSpatial(){
+		return sp;
+	}
+
+	@Override
+	public void simpleUpdate(float tdf) {
+		sp.move(0, 0, 0);
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:49]: This method should be removed as it does nothing.

File: src/main/java/nl/tudelft/contextproject/Bomb.java (hunk 3 above)
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/Bomb.java:54]: There should be some difference between the VRPlayer and the bomb on the generated map.

File: src/test/java/nl/tudelft/contextproject/BombTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Abstract test class for bomb.
+ */
+public abstract class BombTest extends EntityTest{
```
Comment (ChielBruin) [inline src/test/java/nl/tudelft/contextproject/BombTest.java]: This should not be an abstract class.

File: src/test/java/nl/tudelft/contextproject/BombTest.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+	
+	private Bomb bomb;
+	@Before
+	public Drawable getDrawable() {
+		return getBomb();
```
Comment (ChielBruin) [inline src/test/java/nl/tudelft/contextproject/BombTest.java]: return new Bomb();

File: src/test/java/nl/tudelft/contextproject/BombTest.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+	
+	/**
+	 * Test if setting a new state changes the state.
+	 */
+	@Test
+	public void testSetState() {
+		setupbomb();
+		bomb.setState(EntityState.ALIVE);
+		assertEquals(bomb.getState(), EntityState.ALIVE);
+	}
```
Comment (ChielBruin) [inline src/test/java/nl/tudelft/contextproject/BombTest.java:74]: These tests are already inherited from entityTest and are therefore duplicates

**Comment by Diocruel** [comment]
Please keep in mind that failing tests also fail the build. I think we should close the PR until the test have been fixed.

testGetSpatialNotNull(nl.tudelft.contextproject.BombTest)  Time elapsed: 0.009 sec  <<< ERROR!
java.lang.NullPointerException

**Pull request closed without merging**